url284ba87ea1_swabhs_forbidden_paths_tgz-pJ8-tagging_hypergraph_TaggingHypergraphGeneratorJ8/src/hypergraph/HypergraphProto.java

```diff
@@ -0,0 +1,2171 @@
+        return new Builder();
+      }
+      
+      public Builder clear() {
+        super.clear();
+        if (verticesBuilder_ == null) {
+          vertices_ = java.util.Collections.emptyList();
+          bitField0_ = (bitField0_ & ~0x00000001);
+        } else {
+          verticesBuilder_.clear();
+        }
+        if (edgesBuilder_ == null) {
+          edges_ = java.util.Collections.emptyList();
+          bitField0_ = (bitField0_ & ~0x00000002);
+        } else {
+          edgesBuilder_.clear();
+        }
+        return this;
+      }
+      
+      public Builder clone() {
+        return create().mergeFrom(buildPartial());
+      }
+      
+      public com.google.protobuf.Descriptors.Descriptor
+          getDescriptorForType() {
+        return HypergraphProto.Hypergraph.getDescriptor();
+      }
+      
+      public HypergraphProto.Hypergraph getDefaultInstanceForType() {
+        return HypergraphProto.Hypergraph.getDefaultInstance();
+      }
+      
+      public HypergraphProto.Hypergraph build() {
+        HypergraphProto.Hypergraph result = buildPartial();
+        if (!result.isInitialized()) {
+          throw newUninitializedMessageException(result);
+        }
+        return result;
+      }
+      
+      private HypergraphProto.Hypergraph buildParsed()
+          throws com.google.protobuf.InvalidProtocolBufferException {
+        HypergraphProto.Hypergraph result = buildPartial();
+        if (!result.isInitialized()) {
+          throw newUninitializedMessageException(
+            result).asInvalidProtocolBufferException();
+        }
+        return result;
+      }
+      
+      public HypergraphProto.Hypergraph buildPartial() {
+        HypergraphProto.Hypergraph result = new HypergraphProto.Hypergraph(this);
+        int from_bitField0_ = bitField0_;
+        if (verticesBuilder_ == null) {
+          if (((bitField0_ & 0x00000001) == 0x00000001)) {
+            vertices_ = java.util.Collections.unmodifiableList(vertices_);
+            bitField0_ = (bitField0_ & ~0x00000001);
+          }
+          result.vertices_ = vertices_;
+        } else {
+          result.vertices_ = verticesBuilder_.build();
+        }
+        if (edgesBuilder_ == null) {
+          if (((bitField0_ & 0x00000002) == 0x00000002)) {
+            edges_ = java.util.Collections.unmodifiableList(edges_);
+            bitField0_ = (bitField0_ & ~0x00000002);
+          }
+          result.edges_ = edges_;
+        } else {
+          result.edges_ = edgesBuilder_.build();
+        }
+        onBuilt();
+        return result;
+      }
+      
+      public Builder mergeFrom(com.google.protobuf.Message other) {
+        if (other instanceof HypergraphProto.Hypergraph) {
+          return mergeFrom((HypergraphProto.Hypergraph)other);
+        } else {
+          super.mergeFrom(other);
+          return this;
+        }
+      }
+      
+      public Builder mergeFrom(HypergraphProto.Hypergraph other) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 1629-1654 into a separate method.